test/start_time_test.go

```diff
@@ -81,8 +81,8 @@ spec:
 			continue
 		}
 		diff := s.Terminated.StartedAt.Time.Sub(lastStart.Time)
-		if diff < 10*time.Second {
-			t.Errorf("Step %d start time was %s since last start, wanted >10s", idx, diff)
+		if diff < 2*time.Second {
```

Comment:
another nit: 2 is repeated so many times, is it worth putting into a var?

Reply:
One tricky bit here is that `2` is repeated a lot in the YAML and then only twice in the code. Using var substitution in the YAML looks like this:

```go
parse.MustParseTaskRun(`
  steps:
  - image: busybox
    script: sleep %d
  - image: busybox
    script: sleep %d
  - image: busybox
    script: sleep %d
  - image: busybox
    script: sleep %d
  - image: busybox
    script: sleep %d
`, stepSleepSeconds, stepSleepSeconds, stepSleepSeconds, stepSleepSeconds, stepSleepSeconds)
```

Soooo then I write a func to generate the repeated steps and the code becomes this:

```go
parse.MustParseTaskRun(fmt.Sprintf(`
  steps:
%s
`, generateSteps(stepRepeats, stepSleepSeconds))

func generateSteps(repeats, sleepSeconds) string {
  stepYAML = fmt.Sprintf(`
  - image: busybox
    script: sleep %d
`, sleepSeconds)
  return strings.Repeat(stepYAML, repeats)
}
```

And then by this point I can't even read my own code anymore and figure out what it's doing. As gross as the repeated "2"s are I think i'm going to leave them in the YAML and just add a var for the diff check and error message.